OPHD/States/CrimeRateUpdate.cpp

```diff
@@ -0,0 +1,88 @@
+#include "CrimeRateUpdate.h"
+#include "../UI/PopulationPanel.h"
+#include "../Things/Structures/Structure.h"
+#include "../StructureManager.h"
+#include <NAS2D/Utility.h>
+
+namespace CrimeRate
+{
+	bool isProtectedByPolice(const std::vector<TileList>& policeOverlays, Structure* structure);
+	int calculateMoraleChange(int meanCrimeRate);
+	void setPopulationPanel(int moraleChange, int meanCrimeRate, PopulationPanel& populationPanel);
+
+
+	int update(const std::vector<TileList>& policeOverlays, PopulationPanel& populationPanel)
+	{
+		const auto& structuresWithCrime = NAS2D::Utility<StructureManager>::get().structuresWithCrime();
+
+		// Colony will not have a crime rate until at least one structure that supports crime is built
+		if (structuresWithCrime.empty()) {
+			setPopulationPanel(0, 0, populationPanel);
+			return 0;
+		}
+
+		double accumulatedCrime = 0;
+
+		for (auto structure : structuresWithCrime)
+		{
+			int crimeRateChange = isProtectedByPolice(policeOverlays, structure) ? -1 : 1;
+			structure->IncreaseCrimeRate(crimeRateChange);
+
+			accumulatedCrime += structure->crimeRate();
+		}
+
+		int meanCrimeRate = static_cast<int>(accumulatedCrime / structuresWithCrime.size());
+		int moraleChange = calculateMoraleChange(meanCrimeRate);
+
+		setPopulationPanel(moraleChange, meanCrimeRate, populationPanel);
+
+		return moraleChange;
+	}
+
+
+	bool isProtectedByPolice(const std::vector<TileList>& policeOverlays, Structure* structure)
+	{
+		const auto& structureTile = NAS2D::Utility<StructureManager>::get().tileFromStructure(structure);
+
+		for (const auto& tile : policeOverlays[structureTile.depth()])
+		{
+			if (tile->position() == structureTile.position())
+			{
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+
+	int calculateMoraleChange(int meanCrimeRate)
+	{
+		if (meanCrimeRate > 50)
+		{
+			// Reduce morale by 1 for every 10% above 50%
+			return -1 * (meanCrimeRate / 10 - 4);
```

Comment:
I doubt this really matters after optimizations, but you could have just written:
```cpp
return -(meanCrimeRate / 10 - 4);
```